scripts/convert_publish.py

```diff
@@ -0,0 +1,255 @@
+            print(e.args)
+            print(traceback.format_exc())
+        
+        pool["state"]["rshares"] = Decimal128(Decimal(rshares_sum))  # TODO: how should we store int128_t?
+        pool["state"]["rsharesfn"] = Decimal128(Decimal(rshares_sum))# TODO: --//--//--//--//--//--//--//--
+        reward_pools.save(pool)
+        print("...done")
+        
+    def convert_votes(self, query = {}):
+        print("convert_votes")
+        golos_votes = self.golos_db['comment_vote_object']
+        cyberway_messages = self.publish_db['messagetable']
+
+        cursor = golos_votes.find(query)
+        length = cursor.count()
+        print("total votes = ", length)
+        length = golos_votes.count()
+        utils.printProgressBar(0, length, prefix = 'Progress:', suffix = 'Complete', length = 50)
+
+        added = 0
+        passed = -1
+        for doc in cursor:
+            passed += 1
+            try:
+                cur_mssg_id = utils.convert_hash(doc["permlink"])
+                    
+                if (not (doc["author"] in self.exists_accs)) or (not (doc["voter"] in self.exists_accs)):
+                    continue
+
+                vote = {
+                    "id" : added,
+                    "message_id" : cur_mssg_id,
+                    "voter" : doc["voter"],
+                    "weight" : doc["vote_percent"],
+                    "time" : doc["last_update"],
+                    "count" : 0,
+                    "curatorsw": doc["weight"] / 2,
+                    "_SCOPE_" : doc["author"],
+                    "_PAYER_" : "gls.publish",
+                    "_SIZE_" : 50
+                }
+
+                self.publish_tables.vote.append(vote)
+
+                self.mssgs_curatorsw[(doc["author"], cur_mssg_id.bid)] += vote["curatorsw"]
+
+                if added % self.cache_period == 0:
+                    print("votes converted -- ", added)
+                    self.publish_tables.writeCache()
+                added += 1
+                utils.printProgressBar(passed + 1, length, prefix = 'Progress:', suffix = 'Complete', length = 50)
+            except Exception as e:
+                print(doc)
+                print(e.args)
+                print(traceback.format_exc())
+        try:
+            self.publish_tables.writeCache()
+        except Exception as e:
+            print(e.args)
+            print(traceback.format_exc())
+        print('...done')
+    
+    def run(self, query = {}):
+        self.fill_exists_accs()
+        self.convert_votes(query)
+        self.convert_posts(query)
+
+#TODO: config
+MONGO_HOST = "123.123.123.123"
+MONGO_DB = "Golos"
+MONGO_USER = "Jane Doe"
+MONGO_PASS = "Converge2001"
+SSH_USERNAME = "Jane Doe"
+SSH_PASSWORD = "~/.ssh/id_rsa"
+
+def convert(remote, query = {}):
+    client = MongoClient('127.0.0.1', 27017)
+    
+    publish_db = client['_CYBERWAY_gls-publish']
+    cyberway_db = client['_CYBERWAY_TEST_']
+        
+    if remote:
```

Comment:
It looks like this logic should be in a different place. For consistent data transfer, it will be repeated in the script for transferring accounts and witnesses. At the moment, these scripts will be used with databases accessible from one host, so this functionality is not required.

Reply:
U r right. I deleted this logic from the script